Lib9c/Action/ActivateCollection.cs

```diff
@@ -0,0 +1,126 @@
+using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
+using Nekoyume.Module;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [ActionType("activate_collection")]
+    public class ActivateCollection: GameAction
+    {
+        public Address AvatarAddress;
+        public int CollectionId;
+        public List<ICollectionMaterial> Materials = new();
+        public override IWorld Execute(IActionContext context)
+        {
+            context.UseGas(1);
+            var states = context.PreviousState;
+            if (states.TryGetAvatarState(context.Signer, AvatarAddress, out var avatarState))
+            {
+                var sheets = states.GetSheets(containItemSheet: true, sheetTypes: new[]
+                {
+                    typeof(CollectionSheet)
+                });
+                var collectionSheet = sheets.GetSheet<CollectionSheet>();
+                var row = collectionSheet[CollectionId];
+                var materials = Materials;
+                foreach (var materialInfo in row.Materials)
+                {
+                    var material = materials.FirstOrDefault(m =>
+                        m.ItemId == materialInfo.ItemId && m.ItemCount == materialInfo.Count);
+                    if (material is null)
+                    {
+                        throw new Exception();
+                    }
+                    switch (material)
+                    {
+                        case FungibleCollectionMaterial fungibleCollectionMaterial:
+                            if (!avatarState.inventory.RemoveMaterial(materialInfo.ItemId,
+                                    materialInfo.Count))
+                            {
+                                throw new Exception();
+                            }
+                            break;
+                        case NonFungibleCollectionMaterial nonFungibleCollectionMaterial:
+                            var nonFungibleId = nonFungibleCollectionMaterial.NonFungibleId;
+                            if (avatarState.inventory.TryGetNonFungibleItem(nonFungibleId,
+                                    out ItemUsable materialItem) && materialInfo.Validate(materialItem))
+                            {
+                                avatarState.inventory.RemoveNonFungibleItem(materialItem);
+                            }
+                            else
+                            {
+                                throw new Exception();
+                            }
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException(nameof(material));
+                    }
+
+                    materials.Remove(material);
+                }
+
+                if (materials.Any())
+                {
+                    throw new Exception();
+                }
+
+                var collectionAddress = CollectionState.Derive(AvatarAddress);
```

Comment:
We do not more need any `Derive` something. Just like this

`state.GetAccount(CollectionState.StateAddress).GetState(AvatarAddress);`

Now `state.GetAccount(CollectionState.StateAddress)` has its storage system. So you can use the same address for each `Account`. (You can easily notice the `Account` is the same as the `RDBMS` table. but key-value table.)

We strongly recommend separating the state by the domain, like `Avatar` and `Inventory`. So you can search each data with `AvatarAddress` or `AgentAddress` without deriving. Just use it like PK.

Reply:
add commit https://github.com/planetarium/lib9c/pull/2360/commits/1d8adecb1e402fc12c5349d80fa5095ae686152f please check 🙏

Reply:
Much better, but has it passed tests? 

Lib9c/Module/CollectionModule.cs
```
        public static IWorld SetCollectionState(this IWorld world, Address collection, CollectionState state)
        {
            var account = world.GetAccount(Addresses.Collection);
            account = account.SetState(collection, state.SerializeList());
            return world.SetAccount(Addresses.Avatar, account);
        }
```

This should be written state to `Addresses.Collection` I guess...